Darwin/Services/StitchRemoteMongoDBService/StitchSyncPerformanceTests/SyncL2ROnlyPerformanceTestDefinitions.swift

```diff
@@ -0,0 +1,251 @@
+            testName: testName,
+            runId: runId,
+            testDefinition: { (ctx, numDocs, docSize) in
+                // Do the sync pass that will sync the
+                // local changes to the remote collection
+                guard let syncPassSucceeded = try? ctx.dataSynchronizer.doSyncPass(),
+                    syncPassSucceeded else {
+                    // Halt the test if the sync pass failed
+                    fatalError("sync pass failed")
+                }
+        }, beforeEach: { (ctx, numDocs, docSize) in
+            // Generate and insert the documents, and perform the initial sync.
+            print(TAG, "Setting up $testName test for $numDocs $docSize-byte docs")
+            var documentsForCurrentTest =
+                SyncPerformanceTestUtils.generateDocuments(numDoc: numDocs, docSize: docSize)
+
+            // Initial sync for a purely L2R scenario means inserting local documents,
+            // and performing a sync pass to sync those documents up to the remote.
+            let sync = ctx.coll.sync
+
+            // If sync fails for any reason, halt the test
+            sync.configure(
+                conflictHandler: DefaultConflictHandler<Document>.remoteWins(),
+                changeEventDelegate: nil,
+                errorListener: { error, id in
+                    print(
+                        "unexpected sync error with id " +
+                        "\(String(describing: id)): \(error)"
+                    )
+                    fatalError(error.localizedDescription)
+            })
+            sync.insertMany(&documentsForCurrentTest)
+
+            guard let syncPassSucceeded = try? ctx.dataSynchronizer.doSyncPass(),
+                syncPassSucceeded else {
+                // Halt the test if the sync pass failed
+                fatalError("sync pass failed")
+            }
+
+            // Randomly sample a percentage of the documents
+            // that will be locally updated
+            let shuffledDocs = documentsForCurrentTest.shuffled()
+
+            let docsToUpdate = pctOfDocsWithChangeEvents > 0.0 ?
+                shuffledDocs.prefix(upTo: Int(pctOfDocsWithChangeEvents * Double(numDocs))) :
+                []
+
+            docsToUpdate.forEach {
+                _ = sync.updateOne(
+                    filter: ["_id": $0["_id"] ?? BSONNull()],
+                    update: ["$set": ["newField": "blah"] as Document]
+                )
+            }
+
+            numberOfChangedDocs = docsToUpdate.count
+        }, afterEach: { (ctx, numDocs, _) in
+            // Verify that the test did indeed synchronize
+            // the provided documents remotely
+            let numOfDocsSynced = ctx.coll.count(Document())
+            if numDocs != numOfDocsSynced {
+                print(TAG, "\(numDocs) != \(String(describing: numOfDocsSynced))")
+                fatalError("test did not correctly perform the initial sync")
```

Comment:
We want to throw in all of these failures, fatalError() will stop running all of the tests and leave the results document in an unfinished state